src/test/java/io/github/sskorol/testcases/AmazonTests.java

```diff
@@ -2,175 +2,156 @@
 
 import io.github.sskorol.data.Data;
 import io.github.sskorol.data.DataSuppliers;
-import io.github.sskorol.model.*;
+import io.github.sskorol.dataset.*;
+import io.github.sskorol.model.Account;
 import io.github.sskorol.pages.LoginPage;
 import io.github.sskorol.pages.ProductPage;
-import io.github.sskorol.pages.SearchPage;
 import io.qameta.allure.*;
 import org.testng.annotations.Test;
 
 import static io.github.sskorol.assertions.CustomAssertions.customAssertThat;
 import static io.github.sskorol.core.PageFactory.at;
 import static io.github.sskorol.core.PageFactory.open;
+import static io.github.sskorol.model.Category.*;
 import static io.github.sskorol.model.OperationStatus.LOGIN_SUCCESSFUL;
 import static io.github.sskorol.model.OperationStatus.PURCHASE_SUCCESSFUL;
+import static io.github.sskorol.model.SortValues.*;
 
 /**
  * Tests for Amazon.com page.
  */
 public class AmazonTests {
-    @Data(source = "parfume.json", entity = Parfume.class)
+
     @Data(source = "accountAmazon.json", entity = Account.class)
-    @Test(dataProvider = "getDataCollection",
+    @Test(dataProvider = "getObject",
             dataProviderClass = DataSuppliers.class,
-            description = "Should Search For Parfume")
+            description = "Should Login")
     @Feature("Product search")
     @Story("Implement search functionality")
     @Issue("9")
-    @TmsLink("14")
+    @TmsLink("41")
     @Severity(SeverityLevel.BLOCKER)
-    public void shouldSearchForParfume(final Parfume parfume, final Account account) {
+    public void shouldLogIn(final Account account) {
 
         open(LoginPage.class)
                 .login(account.getUsername(), account.getPassword());
 
         customAssertThat(at(LoginPage.class))
                 .hasLoginStatus(LOGIN_SUCCESSFUL);
+    }
 
-        at(SearchPage.class)
-                .searchFor(parfume.getName());
+    @Data(entity = ShouldSearchForParfume.class)
+    @Test(dataProvider = "getDataSet",
+            dataProviderClass = DataSuppliers.class,
+            description = "Should Search For Parfume")
+    @Feature("Product search")
+    @Story("Implement search functionality")
+    @Issue("9")
+    @TmsLink("14")
+    @Severity(SeverityLevel.BLOCKER)
+    public void shouldSearchForParfume(final ShouldSearchForParfume data) {
 
-        at(ProductPage.class)
-                .selectCategoryBy(parfume.getSubCategory())
-                .selectCheckboxBy(parfume.getSize())
-                .sortBy(parfume.getSortValue())
+        open(LoginPage.class)
+                .login(data.getAccount().getUsername(), data.getAccount().getPassword())
+                .searchFor(data.getParfume().getName())
+                .selectCategory(data.getParfume().getSubCategory())
```

Comment:
method assumes `category` arg, but you're passing `subCategory`... logical issue?